oauthenticator/oauth2.py

```diff
@@ -1288,24 +1310,84 @@ async def authenticate(self, handler, data=None, **kwargs):
         """
         # build the parameters to be used in the request exchanging the oauth code for the access token
         access_token_params = self.build_access_tokens_request_params(handler, data)
-        # exchange the oauth code for an access token and get the JSON with info about it
         token_info = await self.get_token_info(handler, access_token_params)
+        # call the oauth endpoints
+        return await self._token_to_auth_model(token_info)
+
+    async def refresh_user(self, user, handler=None, **kwargs):
+        """
+        Renew the Access Token with a valid Refresh Token
+        """
+        if not self.enable_auth_state:
+            # auth state not enabled, can't refresh
+            self.log.debug("auth_state disabled, no auth state to refresh")
+            return True
```

Comment:
Can you update the method docstring with a description of the return value? Returning True when the token isn't refreshed is confusing.